src/app/services/api.service.ts

```diff
@@ -0,0 +1,22 @@
+import { HttpClient } from "@angular/common/http";
+import { Observable } from "rxjs";
+
+export class ApiService {
+    baseURL: string = "SomeURL" // Idk if this is where we wanna have this?
+    header: any; // idk where do to the header and the auth
+
+    constructor(private http: HttpClient){}
```

Comment:
Please provide the http client in the app.config.ts, this will break if we try inject the service without doing this

```ts
provideHttpClient(withFetch(), withInterceptors([requestInterceptor])),
```